src/flp/types/fixedpoint_l2.rs

```diff
@@ -0,0 +1,816 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! A [`Type`] for summing vectors of fixed point numbers whose
+//! [L2 norm](https://en.wikipedia.org/wiki/Norm_(mathematics)#Euclidean_norm) is
+//! bounded by `1`.
+//!
+//! In the following a high level overview over the inner workings of this type
+//! is given and implementation details are discussed. It is not necessary for
+//! using the type, but it should be very helpful when trying to understand the
```

Comment:
nit: The writing style here should be formal(-ish). Avoid using words like "very" that don't add information.